docs/classes/index.rst

```diff
@@ -2,37 +2,137 @@
+
+to access that method we must initialise that class
+
+that can be done like so:
+
+.. code-block:: kotlin
+
+    val a = Example()
+    a.sayHi()
+
+3) Class variables access in Example via class name
+
+In python to access a class property from within that class we use “self”.
+
+In kotlin (from my understanding) there are two ways to access the class properties from within the class.
+
+First one is to..well..just use the property names normally...that was too simple wasn't it?
+
+Second one is to declare a companion object (its properties are private only) and either use the property names as usual, or use the class name in the same way we use “self” in python.
+
+eg
+
+.. code-block:: kotlin
+
+    class Example {
+        companion object {
+            val text = "Hi there!"
+        }
+        
+        fun sayHi() {
+            println(text) // or Example.text
+        }
+    }
 
-    ? - What happens if I don’t do var in the constructor? It’s unresolved later, but where does it go?
 
+4) Constructors && binding constructor arguments to instance attributes (assignment, usage)
 
-In fact, Kotlin has a rich, multi-layered approach to
-construction. Our class attribute ``greeting`` is marked as immutable
-(and *should* be marked with the optional ``private``) as well.
+A class with a constructor in python looks like:
 
-In some ways, Python is clunkier in this example. We have the magic of
-"dunder" names on important methods, such as the "constructor". The symbol
-of ``self`` is sprinkled in to give the instance scope a placeholder. And
-quite obviously, Kotlin's primary constructor -- right after the class
-name -- is terse and doesn't require assigning each value to "self".
+.. code-block:: python
+
+    class Example:
+        def __init__(self, paramOne, paramTwo):
+            self.paramOne = paramOne
+            self.paramTwo = paramTwo
+
+In the above case it sets as properties the two parameters its given.
+
+That can be done in kotlin like this:
+
+.. code-block:: kotlin
+
+    class Example(pOne: String, pTwo: String) {
+        var paramOne = "placeholder"
+        var paramTwo = "placeholder"
+
+        init {
+            paramOne = pOne
+            paramTwo = pTwo
+        }
+        // if you know of a better way please PR this in
+    }
+
+We can then initialise it like this:
+
+.. code-block:: kotlin
+
+    val f = Example("hi", "there")
+    println(f.paramOne)
+
+
+5) Notes
 
 .. note::
 
-    Python's ``__init__`` is called a constructor, but as its name implies,
-    it is actually an initializer. The ``__new__`` method is the factory.
+    ? - What happens if I don’t do var in the constructor? It’s unresolved later, but where does it go?
+
+
+    In fact, Kotlin has a rich, multi-layered approach to construction.
+
+    Our class attribute ``greeting`` is marked as immutable (and *should* be marked with the optional ``private``) as well.
+
+    In some ways, Python is clunkier in this example.
```

Comment:
I agree. Dataclasses are an attempt to de-clunk. Should we just do a mention, or close this section by rewriting the initial Python class as a dataclass?

Reply:
like removing the ``__init__`` method?